### PR TITLE
[Merged by Bors] - chore: adaptations to nightly-2024-03-13

### PR DESCRIPTION
**Pull request by kim-em**
[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Group/Hom/Instances.lean [hunk 1]
```diff
@@ -68,7 +68,9 @@ instance MonoidHom.commGroup {M G} [MulOneClass M] [CommGroup G] : CommGroup (M
       simp,
     zpow_succ' := fun n f => by
       ext x
-      simp [zpow_coe_nat, pow_succ],
+      -- If we combine these two `simp`s we reveal a bug: "unexpected bound variable #0"
```
Comment (jcommelin) [inline Mathlib/Algebra/Group/Hom/Instances.lean:71]: Is there an issue for this? Also, the first simp is now non-terminal.

**Review by jcommelin — COMMENTED** [review]
bors d+

File: Mathlib/Data/Num/Lemmas.lean [hunk 2]
```diff
@@ -266,7 +266,8 @@ theorem ofNat'_succ : ∀ {n}, ofNat' (n + 1) = ofNat' n + 1 :=
 theorem add_ofNat' (m n) : Num.ofNat' (m + n) = Num.ofNat' m + Num.ofNat' n := by
   -- Porting note: `simp` fails to unify `ofNat' (n + 1)` with `ofNat' n.succ`
   have : ∀ {n}, ofNat' n.succ = ofNat' n + 1 := ofNat'_succ
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Num/Lemmas.lean:268]: Still relevant? (Can be followup)

Comment (semorrison) [reply Mathlib/Data/Num/Lemmas.lean]: Cleaned up.

File: Mathlib/Algebra/Group/Hom/Instances.lean (hunk 1 above)
Comment (semorrison) [reply Mathlib/Algebra/Group/Hom/Instances.lean:73]: https://github.com/leanprover-community/mathlib4/issues/11357

Comment (semorrison) [reply Mathlib/Algebra/Group/Hom/Instances.lean:73]: I've also put the link in the comments.

**Pull request closed without merging**